src/lib/FreeC/Backend/Coq/Converter/FuncDecl/Rec/WithSections.hs

```diff
@@ -466,8 +466,8 @@ updateTypeSig mgu constTypeVars argTypeMap returnTypeMap funcDecl = do
   let allTypeArgs = map IR.typeVarIdent typeArgVars
       entry'      = entry { entryArity         = length args
                           , entryTypeArgs      = allTypeArgs \\ constTypeVars
-                          , entryArgTypes      = argTypes
-                          , entryReturnType    = returnType
+                          , entryArgTypes      = map fromJust argTypes
```

Comment:
> I'm assuming that `fromJust` is needed due to the `Map.lookup` operation yielding `Maybe`. Would it make sense to switch to a total map instead?

The above comment was referring to this line since `argTypes` is built using `Map.lookup`.

Reply:
Do you mean changing `argTypeMap` to a map that is total by having a default value? I don't think that would be useful here, because it would be difficult to come up with a useful default value other than `error`.
The other kind of total maps I have seen only work with Enum-types as keys, which we definitely can't use here.

However, `fromJust` could be avoided by using the [`(!)`](https://hackage.haskell.org/package/containers-0.6.2.1/docs/Data-Map-Strict.html#v:-33-)-operator instead of `lookup` to build `argTypes` and `returnType`. This would slightly reduce the length of the code, but the location where errors potentially occur would be less explicit, so I don't know if this change is desired.

Reply:
I see. The default value I had in mind was indeed `error`. If we assume that the lookup never fails and, thus, we do not need the ability to distinguish between success and failure at the call site, I would prefer a total map. In addition, moving `error` to the map would allow adding the key to the error message. However, I am not sure whether this information is actually helpful.

Reply:
Okay. But to add the key to the error message, we would need a default function of type `k -> a` instead of a default value of type `a` (assuming a map of type `Map k a`). The packages providing data types for total maps I have found (like [this](https://hackage.haskell.org/package/total-map)) only use default values instead of functions.
I would therefore suggest to avoid using an additional package and changing the type of the map itself, but instead use a new `lookup`-like function that uses `Map.lookup` and returns the result of this lookup without `Maybe` or `error` with an error message containing the key.
(The existing function `Map.findWithDefault` can't be used because it too works with default values and not functions).

Reply:
Since the code in this module is very lengthy already, I'd argue against complicating it even more by tweaking some internal error messages. The use of `fromJust` looks safe to me in this scenario. I think that it is more important that fields of environment entries are evaluated strictly such that potential errors are thrown in the first place. However, that's beyond the scope of issue #19.

Reply:
So the information about the key and map for which the lookup failed is not useful? My issue with `fromJust` is that you essentially lose all information associated with how the error occurred and only know that _something_ went wrong. But this is probably a discussion worth having in its own issue, I'll mark this as resolved here.